api/manifest.go

```diff
@@ -243,7 +242,6 @@ func readManifest(mr storage.LazySectionReader, addr storage.Address, fileStore
 	if err != nil { // size == 0
 		// can't determine size means we don't have the root chunk
 		log.Trace("manifest not found", "addr", addr)
-		err = fmt.Errorf("Manifest not Found")
```

Comment:
leaving a comment here mainly for @zelig:

the `Manifest not Found` error created here was swallowing up the `420` that was trying to be returned.

in any case, i believe without line `246` the returned error will be `500` or something of the sort (@santicomp2014 can confirm this)

Reply:
Yes exactly, the error code becomes Manifest not found and the error code is 500.
The error `manifest not found` is already being thrown in the error.